control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -980,49 +888,60 @@ impl CreateVolumeExe for CreateVolume {
     async fn create<'a>(
         &'a self,
         context: &mut Context<'a>,
-        candidates: CreateReplicaCandidate,
+        candidates: Self::Candidates,
     ) -> Vec<Replica> {
-        let mut replicas = Vec::<Replica>::with_capacity(candidates.candidates().len());
-        for replica in candidates.candidates() {
-            if replicas.len() >= self.replicas as usize {
-                break;
-            } else if replicas.iter().any(|r| {
-                r.node == replica.node
-                    || spread_label_is_same(r, replica, context).unwrap_or_else(|error| {
+        let mut fina_replicas = Vec::<Replica>::new();
+        for crc in candidates.iter() {
+            let mut replicas = Vec::<Replica>::with_capacity(crc.candidates().len());
```

Comment:
We don't seem to be considering replicas which are already part of the volume, example from previous creation attempts?
How does this work and do they affect the creation of the candidates?

Reply:
For every set of create replica candidates  , I am checking if the required number of replica is sufficed or not.

Let say  Set[[PoolA,PoolB, PoolC], [PoolD, PoolE]] 
If PoolA, PoolB , PoolC are good enough and replica can be created then , create the replica, dont look for PoolD, PoolE.
If the first element of set i.e. PoolA, PoolB, PoolC and accomodate the replicas, even if the first replica was present on PoolA, we dont check it. 

We just skip this set and move on to other i.e. PoolD and PoolE